packages/dds/map/src/map.ts

```diff
@@ -334,7 +334,10 @@ export class SharedMap extends SharedObject<ISharedMapEvents> implements IShared
 		const json = await readAndParse<object>(storage, snapshotFileName);
```

Comment:
See https://github.com/microsoft/FluidFramework/pull/19242#discussion_r1484715046

I think it'd be better to do `readBlob` then `this.serializer.parse`, to be consistent that handles are always unencoded in this code.